src/common/acl_principal.c

```diff
@@ -0,0 +1,311 @@
+/*
+ * (C) Copyright 2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+/*
+ * daos_acl: This file contains functions related to working with the principals
+ * from Access Control Lists.
+ */
+
+#include <daos_security.h>
+#include <gurt/common.h>
+#include <pwd.h>
+#include <grp.h>
+
+#define DEFAULT_BUF_LEN 1024
+
+/*
+ * No platform-agnostic way to fetch the max buflen - so let's try a
+ * sane value and double until it's big enough.
+ *
+ * Assumes expr is part of the getpw*_r family of functions, and rc/buf/buflen
+ * and a goto label "out" are already defined in the surrounding function.
+ */
+#define TRY_UNTIL_BUF_SIZE_OK(expr)				\
+{								\
```

Comment:
do {
...
} while (0)

This makes it so you can place it anywhere and it t will behave like a single statement.  As written, if you do something like 

if (...)
   Macro(...);
else
   ...

You’ll get compiler error because of two statements after if.

Also, it would be better to move local vars into this clock (like buflen).

Reply:
Fixed.